File: day25-Running Time and Complexity/Solution.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        
        for (int i=0; i<T; i++) 
        {
            int N = sc.nextInt();
            // Fast check if number is divisible by 2
            if (N < 2 || (N > 2 && (N&1) == 0)) System.out.println("Not prime");
            else
            {
                boolean isPrime = true;
                for (int j = 2; j < (int)Math.sqrt(N); j++) 
                {
                    if (N%j == 0) 
                    {
                        isPrime = false;
                        break;
                    }
                }
                System.out.println(isPrime?"Prime":"Not prime");
            }
        }
    }
}

File: day25-Running Time and Complexity/Solution.c
#include <stdio.h>
#include <math.h>

int main() 
{
    int T, N, isPrime;
    scanf("%d", &T);
    
    for (int i = 0; i < T; i++) 
    {
        scanf("%d", &N);
        
        // Fast check if number is divisible by 2
        if (N < 2 || (N > 2 && !(N&1))) printf("Not prime\n");
        else 
        {
            isPrime = 1;
            for (int j = 2; j < (int)sqrt(N); j++) 
            {
                if (!(N%j)) 
                {
                     isPrime = 0;
                     break;
                }
            }
            printf(isPrime?"Prime\n":"Not prime\n");
        }
    }
    return 0;
}